go-controller/pkg/factory/factory.go

```diff
@@ -96,7 +108,27 @@ func NewWatchFactory(ovnClientset *util.OVNClientset) (*WatchFactory, error) {
 		return nil, err
 	}
 
-	// Create shared informers we know we'll use
+	// For Services and Endpoints, pre-populate the shared Informer with one that
+	// has a label selector excluding headless services.
```

Comment:
I guess that means we could get rid of the checks we have in different service handler code paths where we check if it is a headless service?

Reply:
thinking out load, what happens with services that move from headless to clusterIP or vice versa?
is not common, but is a possibility

Reply:
Then the label selector will match or not match, and the apiserver (should) generate the correct ADD or DELETE event.

This is exactly the selector that kube-proxy uses, so I assume this works.

Reply:
ic, IsHeadlessService label applies to endpoints, so is safe to discard endpoints that belongs to headless services.

https://github.com/kubernetes/kubernetes/blob/42273a4dbf27cd2ed362c1b2e418d2e15fe2c53c/staging/src/k8s.io/api/core/v1/well_known_labels.go#L45-L49

Reply:
@squeed what about my question?

Reply:
Yeah, it turns out I wasn't quite right with how the controller works - the headless label is only set on Endpoints and EndpointSlices.

I'll update this code to only add this selector to Endpoints.

Reply:
In other words, I assume we still need those checks for the Service controller, but can skip it for Endpoints (if we have any)